File: escenario3_docker-tcp-python/benchmark.py
# benchmark.py
import socket
import time
import statistics

HOST = "127.0.0.1"   # dirección del servidor
PORT = 8080          # puerto del servidor
ITERACIONES = 100

latencias = []

for i in range(ITERACIONES):
    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
        s.connect((HOST, PORT))
        start = time.time()
        s.sendall(b"estimulo")
        data = s.recv(1024)
        elapsed = (time.time() - start) * 1000  # convertir a ms
        latencias.append(elapsed)

        # opcional: imprimir cada resultado
        print(f"Iteración {i+1}: {elapsed:.3f} ms - Respuesta: {data.decode()}")

# calcular métricas
promedio = statistics.mean(latencias)
minimo = min(latencias)
maximo = max(latencias)

print("\n📊 Resultados del benchmark")
print(f"Solicitudes: {ITERACIONES}")
print(f"Latencia promedio: {promedio:.3f} ms")
print(f"Latencia mínima: {minimo:.3f} ms")
print(f"Latencia máxima: {maximo:.3f} ms")

File: escenario1_docker-tcp-go/benchmark.go
// benchmark.go
package main

import (
    "fmt"
    "net"
    "time"
)

func main() {
    host := "127.0.0.1:8080"
    iteraciones := 100
    latencias := make([]float64, 0, iteraciones)

    for i := 0; i < iteraciones; i++ {
        conn, err := net.Dial("tcp", host)
        if err != nil {
            panic(err)
        }

        start := time.Now()
        conn.Write([]byte("estimulo"))
        buffer := make([]byte, 1024)
        n, _ := conn.Read(buffer)
        elapsed := time.Since(start).Seconds() * 1000 // ms
        conn.Close()

        latencias = append(latencias, elapsed)
        fmt.Printf("Iteración %d: %.3f ms - Respuesta: %s\n", i+1, elapsed, string(buffer[:n]))
    }

    // calcular métricas
    var suma, min, max float64
    min, max = latencias[0], latencias[0]
    for _, v := range latencias {
        suma += v
        if v < min {
            min = v
        }
        if v > max {
            max = v
        }
    }
    promedio := suma / float64(iteraciones)

    fmt.Println("\n📊 Resultados del benchmark")
    fmt.Printf("Solicitudes: %d\n", iteraciones)
    fmt.Printf("Latencia promedio: %.3f ms\n", promedio)
    fmt.Printf("Latencia mínima: %.3f ms\n", min)
    fmt.Printf("Latencia máxima: %.3f ms\n", max)
}
